test/k8sT/assertionHelpers.go

```diff
@@ -58,7 +58,7 @@ func ExpectCiliumNotRunning(vm *helpers.Kubectl) {
 // the error returned by that function is nil.
 func ExpectCiliumOperatorReady(vm *helpers.Kubectl) {
 	By("Waiting for cilium-operator to be ready")
-	err := vm.WaitforPods(helpers.CiliumNamespace, "-l name=cilium-operator", longTimeout)
+	err := vm.WaitForSinglePod(helpers.CiliumNamespace, "-l name=cilium-operator", longTimeout)
```

Comment:
Shouldn't we check if it's a single node deployment, and call `WaitForSinglePod`/`WaitForPods` accordingly?

Reply:
Maybe, but isn't the whole point of having two operator instances HA, i.e., having one of them operational should be fine? Unless we explicitly want to test that in a specific scenario both of them come up?

Reply:
It's possible that we make a change in the future where the second operator instance fails and this wouldn't alert anyone that something unexpected happened. I think I would agree with Aditi.

Reply:
If we make any features that require operator to have two running instances for some reason, it should be tested separately as a separate e2e suite. I think this change is fine as it is.

Reply:
Changed to retain the old behavior if number of k8s nodes != 1.